Figure

MashaChernova opened the pull request:
При создании фигуры проверяется являются ли стороны числовыми значениями больше нуля (для этого реализован отдельный класс Sides), от абстрактного класса Figure наследуются Triangle, Rectangle, Circle, от Rectangle наследуется Squere. При инициализации треугольника проверяется выполнение неравенства треугольника, если он не существуют, выбрасывается reise ValueErrow

MashaChernova committed:
Классы геометрических фигур. Есть проверка на числовое значение сторон, существование (для треугольника) методы вычисления площадей и периметров. Кроме того имеется метод, который добавляет к площади одной фигуры площадь другой.

MashaChernova committed:
Классы геометрических фигур

PavelHightTower commented on src/Sides.py:
```diff
@@ -0,0 +1,11 @@
+        self.side_lenth = side_lenth
+
+    @property
+    def isCorrectSide(self) :
+        if not (isinstance(self.side_lenth,int) or isinstance(self.side_lenth,float)) :
+            raise ValueError("Введено не число")
```
тут нужно быть осторожно, что можно передать True

и в целом создание класса выглядит как переусложнение

можно было функцию валидатор написать

def side_validation(*args):
       тут в цикле проходить по сторонам и проверять

MashaChernova committed:
тесты, проверяющие требования к программе:
У каждой фигуры есть методы вычисления площади и периметра
Если треугольник не существуют, то появляется ошибка
Для каждой функции задана функция, добавляющая площадь другой фигуры
и если в нее подана не фигура, то выдается ошибка ValueError

e7506c commented on tests/test_add_area_positive.py:
```diff
@@ -0,0 +1,15 @@
+                          (Circle(1.0),Square(2.0)),
+                          (Rectangle(1,2),Square(3))],
+                         ids=["Squere_Rectangle", "Triangle_Triangle", "Circle_Squere", "Rectangle_Squere"])
+def test_circle_area(figure1, figure2):
```
название теста немного не отображает суть

e7506c commented on tests/test_circle_positiv.py:
```diff
@@ -0,0 +1,19 @@
+                         ids=["integer", "float"])
+def test_circke_perimeter(radius, perimeter):
+    с = Circle(radius=radius)
+    assert с.perimeter == perimeter, f'Периметр должен быть 12, а он {с.perimeter}'
```
а почему 12?)

e7506c commented on tests/test_triangle_not_exist.py:
```diff
@@ -0,0 +1,8 @@
+
+@pytest.mark.xfail(raises = ValueError)
+def test_circle_area(side_a=1,side_b=2, side_c=4):
+    Triangle(11,2,5)
```
тут лучше использовать [pytest.raises](https://docs-python.ru/packages/frejmvork-pytest/funktsija-pytest-raises/)
